apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -763,7 +763,7 @@ enqueue_batch(IsReplay, BatchSize, Srs0, Session = #{inflight := Inflight0}, Cli
             true -> ItBegin0;
             false -> ItEnd0
         end,
-    case emqx_ds:next(?PERSISTENT_MESSAGE_DB, ItBegin, BatchSize) of
+    case emqx_ds:next(?PERSISTENT_MESSAGE_DB, ItBegin, BatchSize, #{use_cache => true}) of
```

Comment:
Why do we need a new option for the cache? Does it change the semantics of `next`?

Reply:
Semantics, no.  But we need a way to call next without going through the cache.  One example is the cache worker: when it tries to update the cache, it should go through it.

Reply:
This could be solved by having an internal API in the replication layer that always goes directly to the RPC/RocksDB, and exposing the cached version to `emqx_ds` callback module (similar to how egress workers intercept calls).

Exposing this option directly in the API only makes sense when we want to give the API consumers an option to bypass cache, but I can't come up with a situation when it would be needed.

Reply:
If the cache is to be used by other backends, then shouldn't it be controllable from `emqx_ds`, like egress is as well?

Reply:
Egress also has its options exposed:

https://github.com/emqx/emqx/blob/ed8660cb681a03a07b6d4f5faa59dc35a9518314/apps/emqx_durable_storage/src/emqx_ds.erl#L251-L257

Reply:
Inverted the default behavior to use the cache by default in the replication layer.

Reply:
Options exposed by egress actually change the semantics of the call: for example, `async => true` means the message can be lost before being persisted.

Reply:
Ok.  Still, if I understand correctly, the cache could be used by other backends.  If so, the way to call `next` without attempting to use the case should live in `emqx_ds`, right?